src/passes/SignaturePruning.cpp

```diff
@@ -149,6 +150,12 @@ struct SignaturePruning : public Pass {
       }
     }
 
+    // Types with subtypes cannot be pruned: for a function type B to be a
+    // subtype of A, we need them to have the same number of parameters, as B's
+    // parameters must be a subtype of A's, and tuple types are only subtypes if
+    // their sizes match.
```

Comment:
```suggestion
    // Types with subtypes cannot be pruned: for a function type B to be a
    // subtype of A, we need them to have the same number of parameters, as B's
    // parameters must be supertypes of A's, and tuple types are only subtypes if
    // their sizes match.
```
Due to contravariance of parameters ;) Although we could also simplify this whole comment to just say the arities have to match.

Reply:
Does contravariance always mean the number is the same?

Reply:
No, not by itself.

Reply:
Then should the comment not only mention covariance and contravariance but also the number?